draft-ietf-quic-transport.md

```diff
@@ -941,17 +941,14 @@ connection ID for the duration of the connection or until its peer invalidates
 the connection ID via a RETIRE_CONNECTION_ID frame
 ({{frame-retire-connection-id}}).
 
-Endpoints store received connection IDs for future use.  An endpoint that
-receives excessive connection IDs MAY discard those it cannot store without
-sending a RETIRE_CONNECTION_ID frame.  An endpoint that issues connection IDs
-cannot expect its peer to store and use all issued connection IDs.
-
 An endpoint SHOULD ensure that its peer has a sufficient number of available and
-unused connection IDs.  While each endpoint independently chooses how many
-connection IDs to issue, endpoints SHOULD provide and maintain at least eight
-connection IDs.  The endpoint SHOULD do this by always supplying a new
-connection ID when a connection ID is retired by its peer or when the endpoint
-receives a packet with a previously unused connection ID.  Endpoints that
+unused connection IDs. Endpoints store received connection IDs for future use.
+An endpoint uses a transport parameter to advertise the maximum number of
+connection IDs it can store for future use. An endpoint MUST NOT provide more
+connection IDs than this limit. An endpoint MUST treat receipt of more than this
+number of connection IDs as an error of type CONNECTION_ID_LIMIT_ERROR. If an
```

Comment:
Do we want a new error for this, or can we live with PROTOCOL_VIOLATION?

Reply:
That's the old discussion of error codes vs. error messages, right? Considering the fact that we have separate *_LIMIT_ERRORs for flow control and stream IDs, I thought it would be consistent to have an error code for this as well. I could live with PROTOCOL_VIOLATION as well, if people are unhappy with that.